src/shared/components/iframe-runtime.tsx

```diff
@@ -69,10 +67,17 @@ export const IframeRuntime: React.FC<IProps> =
         }
         return Promise.resolve({});
       });
-
       phone.addListener("interactiveState", (newInteractiveState: any) => {
         setInteractiveStateRef.current?.(newInteractiveState);
-        resolveOnUnload.current?.(newInteractiveState);
+        if (onUnloadCallback && resolveOnUnload.current) {
+          // send the interactive state to any parent interactive that has provided an onUnload 
+          // callback, and then resolve the promise that was saved in the setOnUnload function 
+          // with undefined so the parent interactive doesn't have its state overwritten.
+          onUnloadCallback(newInteractiveState);
+          resolveOnUnload.current(undefined);
+        } else {
+          resolveOnUnload.current?.(newInteractiveState);
+        }
```

Comment:
This is not really about this change but I'm wondering though if we should set `resolveOnUnload.current` to `null` here so that we are guaranteed to only call it once.

Reply:
@dougmartin Thanks. I've made that change and it seems to work fine. Look good to you?